presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -0,0 +1,268 @@
+                .stream()
+                .filter(entry -> !entry.getValue().isDone())
+                .map(entry -> entry.getKey().getQueryId())
+                .distinct()
+                .collect(toImmutableList());
+
+        for (QueryId queryId : queriesWithDynamicFilters) {
+            Supplier<List<StageInfo>> stageInfoSupplier = queries.get(queryId);
+            if (stageInfoSupplier != null) {
+                for (StageInfo stageInfo : getStagesWithDynamicFilters(stageInfoSupplier.get())) {
+                    StageState stageState = stageInfo.getState();
+                    // wait until stage has finished scheduling tasks
+                    if ((stageState == SCHEDULING_SPLITS || stageState == SCHEDULED || stageState == RUNNING || stageState == FINISHED)) {
+                        List<TaskInfo> tasks = stageInfo.getTasks();
+                        Map<String, ImmutableList<TupleDomain<String>>> stageDynamicTupleDomains = tasks.stream()
+                                .map(TaskInfo::getDynamicTupleDomains)
+                                .flatMap(taskDynamicTupleDomains -> taskDynamicTupleDomains.entrySet().stream())
+                                .collect(groupingBy(Map.Entry::getKey, mapping(Map.Entry::getValue, toImmutableList())));
+                        for (Map.Entry<String, ImmutableList<TupleDomain<String>>> entry : stageDynamicTupleDomains.entrySet()) {
+                            List<TupleDomain<String>> sourceDynamicTupleDomains = entry.getValue();
+                            // check if all tasks of a dynamic filter source have reported TupleDomain summary
+                            if (tasks.size() == sourceDynamicTupleDomains.size()) {
+                                dynamicFilterSummaries.computeIfAbsent(
+                                        SourceDescriptor.of(queryId, entry.getKey()),
+                                        ignored -> SettableFuture.create())
+                                        .set(columnWiseUnion(sourceDynamicTupleDomains));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public ListenableFuture<TupleDomain<String>> getSummary(QueryId queryId, String source)
+    {
+        return dynamicFilterSummaries.computeIfAbsent(SourceDescriptor.of(queryId, source), ignored -> SettableFuture.create());
+    }
+
+    public void removeQuery(QueryId queryId)
+    {
+        dynamicFilterSummaries.keySet().removeIf(sourceDescriptor -> sourceDescriptor.getQueryId().equals(queryId));
+        queries.remove(queryId);
+    }
+
+    public Supplier<TupleDomain<ColumnHandle>> createDynamicFilterSupplier(QueryId queryId, List<DynamicFilters.Descriptor> dynamicFilters, Map<Symbol, ColumnHandle> columnHandles, Map<String, PlanNodeId> dynamicFilterSources)
+    {
+        AtomicReference<TupleDomain<ColumnHandle>> dynamicPredicates = new AtomicReference<>(TupleDomain.all());
```

Comment:
build a mapping from `String` to `ColumnHandle` here instead of `translateSummaryToTupleDomain`

Reply:
made changes to build mapping here